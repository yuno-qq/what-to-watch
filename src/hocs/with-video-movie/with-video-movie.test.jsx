import React from 'react'
import {describe, test, expect} from "@jest/globals"
import renderer from 'react-test-renderer'

import withVideoMovie from "./with-video-movie.jsx"


describe(`<withVideoMovie> snapshots:`, () => {
  test(`should render component with video MovieCards`, () => {
    const MockComponentWrapped = withVideoMovie((props) => {
      const {
        renderItems
      } = props

      const movies = [
        {
          genre: {
            id: `comedies`,
            name: `Comedies`,
          },
          imageSrc: `https://placehold.co/280x175/EEE/31343C`,
          url: `/`,
          previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
          name: `Красный дракон`
        },
        {
          genre: {
            id: `comedies`,
            name: `Comedies`,
          },
          id: `111111`,
          imageSrc: `https://placehold.co/280x175/EEE/31343C`,
          url: `/`,
          previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
          name: `Молчание ягнят`
        },
      ]

      return (
        <div className="MOCK_WRAPPER">
          {renderItems(movies, movies.length)}
        </div>
      )
    })

    const tree = renderer.create(<MockComponentWrapped />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
