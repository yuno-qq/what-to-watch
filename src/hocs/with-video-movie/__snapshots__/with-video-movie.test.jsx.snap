// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<withVideoMovie> snapshots: should render component with video MovieCards 1`] = `
<div
  className="MOCK_WRAPPER"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        loop={true}
        muted={true}
        onCanPlay={[Function]}
        onWaiting={[Function]}
        poster="https://placehold.co/280x175/EEE/31343C"
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/"
      >
        Красный дракон
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        loop={true}
        muted={true}
        onCanPlay={[Function]}
        onWaiting={[Function]}
        poster="https://placehold.co/280x175/EEE/31343C"
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/"
      >
        Молчание ягнят
      </a>
    </h3>
  </article>
</div>
`;
